designs/initial.md

```diff
@@ -0,0 +1,56 @@
+## `@rest` directive
+
+#### arguments:
+- route: path to rest endpoint. This could be a path or a full url. If a path, add to the endpoint given on link creation or from the context
+- provides: a map of variables to url params
+- method: the HTTP method to send the request via (i.e GET, PUT, POST)
+- type: The GraphQL type this will return
```

Comment:
What does this type mean? Do we also need to declare a schema for all the REST data?

Reply:
it is what `__typename` would return if this was a GraphQL endpoint. My thought was to make this ad-hoc instead of a schema

Reply:
I guess I'm not really sure what it would be used for

Reply:
@jbaxleyiii -- Won't this be dangerous if we don't have a proper schema for the response?